primedev/plugins/plugins.cpp

```diff
@@ -1,340 +1,235 @@
+
+	if (!logName)
 	{
-		NS::log::PLUGINSYS->error(
-			"'{}' has an incompatible API version number in its manifest. Current ABI version is '{}'", pathstring, ABI_VERSION);
-		return std::nullopt;
+		NS::log::PLUGINSYS->error("Could not load logName of plugin {}", name);
+		return;
 	}
-	// Passed all checks, going to actually load it now
 
-	HMODULE pluginLib =
-		LoadLibraryExW(wpptr, 0, LOAD_LIBRARY_SEARCH_USER_DIRS | LOAD_LIBRARY_SEARCH_DEFAULT_DIRS); // Load the DLL with lib folders
-	if (pluginLib == NULL)
+	if (!dependencyName)
 	{
-		NS::log::PLUGINSYS->info("Failed to load library '{}': ", std::system_category().message(GetLastError()));
-		return std::nullopt;
+		NS::log::PLUGINSYS->error("Could not load dependencyName of plugin {}", name);
+		return;
 	}
-	plugin.init = (PLUGIN_INIT_TYPE)GetProcAddress(pluginLib, "PLUGIN_INIT");
-	if (plugin.init == NULL)
+
+	if (!isValidSquirrelIdentifier(this->dependencyName))
 	{
-		NS::log::PLUGINSYS->info("Library '{}' has no function 'PLUGIN_INIT'", pathstring);
-		return std::nullopt;
+		NS::log::PLUGINSYS->error("Dependency name \"{}\" of plugin {} is not valid", dependencyName, name);
+		return;
 	}
-	NS::log::PLUGINSYS->info("Succesfully loaded {}", pathstring);
-
-	plugin.name = manifestJSON["name"].GetString();
-	plugin.displayName = manifestJSON["displayname"].GetString();
-	plugin.description = manifestJSON["description"].GetString();
-	plugin.api_version = manifestJSON["api_version"].GetString();
-	plugin.version = manifestJSON["version"].GetString();
-
-	plugin.run_on_client = manifestJSON["run_on_client"].GetBool();
-	plugin.run_on_server = manifestJSON["run_on_server"].GetBool();
 
-	if (!plugin.run_on_server && IsDedicatedServer())
-		return std::nullopt;
+	this->callbacks = (IPluginCallbacks*)CreatePluginInterface("PluginCallbacks001", 0);
 
-	if (manifestJSON.HasMember("dependencyName"))
+	if (!this->callbacks)
 	{
-		plugin.dependencyName = manifestJSON["dependencyName"].GetString();
+		NS::log::PLUGINSYS->error("Could not create callback interface of plugin {}", name);
+		return;
 	}
-	else
+
+	this->logger = std::make_shared<ColoredLogger>(this->logName, NS::Colors::PLUGIN);
+	RegisterLogger(this->logger);
+
+	if (IsDedicatedServer() && !this->runOnServer)
 	{
-		plugin.dependencyName = plugin.name;
+		NS::log::PLUGINSYS->error("Plugin {} did not request to run on dedicated servers", this->name);
+		return;
 	}
 
-	if (std::find_if(
-			plugin.dependencyName.begin(),
-			plugin.dependencyName.end(),
-			[&](char c) -> bool { return !((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || c == '_'); }) !=
-		plugin.dependencyName.end())
+	if (!IsDedicatedServer() && !this->runOnClient)
 	{
-		NS::log::PLUGINSYS->warn("Dependency string \"{}\" in {} is not valid a squirrel constant!", plugin.dependencyName, plugin.name);
+		NS::log::PLUGINSYS->warn("Plugin {} did not request to run on clients", this->name);
```

Comment:
isnt this more "non dedi" rather than "clients"

Reply:
Honestly maybe the  "context" if plugins are running on a dedi or client should get the boot entirely.